gc_registry/tests/conftest.py

```diff
@@ -321,3 +322,48 @@ def fake_db_solar_device(
     )
 
     return device_read
+
+
+@pytest.fixture()
+def fake_db_gc_bundle(
+    db_write_session: Session, db_read_session: Session, fake_db_wind_device: Device
+) -> GranularCertificateBundle:
+    gc_bundle_dict = {
+        "account_id": 1,
+        "certificate_status": "Active",
+        "bundle_id_range_start": 0,
+        "bundle_id_range_end": 1000,
+        "bundle_quantity": 1001,
+        "energy_carrier": EnergyCarrierType.electricity,
+        "energy_source": EnergySourceType.wind,
+        "face_value": 1,
+        "issuance_post_energy_carrier_conversion": False,
+        "registry_configuration": 1,
+        "device_id": fake_db_wind_device.id,
+        "device_name": "fake_wind_device",
```

Comment:
Shouldn't need to provide these device attributes to the GCB any more after normalisation

Reply:
See I thought this as well but the `model_validate` underneath thrown a strop if all of the features aren't in the dict. Do you reckon another step is needed perhaps? In the `split_bundle` branch I actually need to also init an `issuance_metadata` entity beforehand so that the foreign key as something to cling to

Reply:
I just getting validation errors for: `is_storage`, `issuance_id` and `metadata_id` after commenting out the attributes for metadata and device.

i.e.
```
    gc_bundle_dict = {
        "account_id": 1,
        "certificate_status": "Active",
        "bundle_id_range_start": 0,
        "bundle_id_range_end": 1000,
        "bundle_quantity": 1001,
        "energy_carrier": EnergyCarrierType.electricity,
        "energy_source": EnergySourceType.wind,
        "face_value": 1,
        "issuance_post_energy_carrier_conversion": False,
        "registry_configuration": 1,
        "device_id": fake_db_wind_device.id,
        "production_starting_interval": "2021-01-01T00:00:00",
        "production_ending_interval": "2021-01-01T01:00:00",
        "issuance_datestamp": "2021-01-01",
        "expiry_datestamp": "2024-01-01",
        "emissions_factor_production_device": 0.0,
        "emissions_factor_source": "Some Data Source",
    }
```

Reply:
Try now

Reply:
Validation passes, but I'm getting this error. Do you want to add a basic test to the branch and check if you have the same issue:

```
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column "is_deleted" of relation "granularcertificatebundle" does not exist
E       LINE 1: ...ge, sdr_allocation_id, storage_efficiency_factor, is_deleted...
E                                                                    ^
E       
E       [SQL: INSERT INTO granularcertificatebundle (issuance_id, hash, certificate_status, account_id, metadata_id, bundle_id_range_start, bundle_id_range_end, bundle_quantity, energy_carrier, energy_source, face_value, issuance_post_energy_carrier_conversion, emissions_factor_production_device, emissions_factor_source, device_id, production_starting_interval, production_ending_interval, expiry_datestamp, is_storage, sdr_allocation_id, storage_efficiency_factor, is_deleted, created_at, id) VALUES (%(issuance_id)s, %(hash)s, %(certificate_status)s, %(account_id)s, %(metadata_id)s, %(bundle_id_range_start)s, %(bundle_id_range_end)s, %(bundle_quantity)s, %(energy_carrier)s, %(energy_source)s, %(face_value)s, %(issuance_post_energy_carrier_conversion)s, %(emissions_factor_production_device)s, %(emissions_factor_source)s, %(device_id)s, %(production_starting_interval)s, %(production_ending_interval)s, %(expiry_datestamp)s, %(is_storage)s, %(sdr_allocation_id)s, %(storage_efficiency_factor)s, %(is_deleted)s, %(created_at)s, %(id)s)]
E       [parameters: {'issuance_id': '\n        1-         EnergyCarrierType.electricity-         2021-01-01T00:00:00\n        ', 'hash': '4ac9465415080bcdd0dfe2659870a24264617faf29361415cb8d44899c9e9389', 'certificate_status': 'ACTIVE', 'account_id': 1, 'metadata_id': 1, 'bundle_id_range_start': 0, 'bundle_id_range_end': 999, 'bundle_quantity': 1000, 'energy_carrier': 'electricity', 'energy_source': 'wind', 'face_value': 1, 'issuance_post_energy_carrier_conversion': False, 'emissions_factor_production_device': 0.0, 'emissions_factor_source': 'Some Data Source', 'device_id': 1, 'production_starting_interval': datetime.datetime(2021, 1, 1, 0, 0), 'production_ending_interval': datetime.datetime(2021, 1, 1, 1, 0), 'expiry_datestamp': datetime.datetime(2024, 1, 1, 0, 0), 'is_storage': 0, 'sdr_allocation_id': None, 'storage_efficiency_factor': None, 'is_deleted': False, 'created_at': datetime.datetime(2024, 10, 14, 11, 6, 43, 733309), 'id': 1}]
E       (Background on this error at: https://sqlalche.me/e/20/f405)
```

I just added this to test the fixture:

```
def test_gcb_creation(fake_db_gc_bundle):

    assert fake_db_gc_bundle.id is not None
```